// mui-react/lib/components/Backdrop.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * @file This file contains Backdrop component, which is used in Modal and Picker
                                                                                                                                                                                                                              * @author mayingcong <mayingcong@xingxy.cn>
                                                                                                                                                                                                                              * @date 2016-11-10
                                                                                                                                                                                                                              */

var Backdrop = _react2.default.createClass({
    displayName: 'Backdrop',

    propTypes: {
        isOpen: _react2.default.PropTypes.bool
    },
    getDefaultProps: function getDefaultProps() {
        return {
            isOpen: false
        };
    },

    render: function render() {
        var _props = this.props;
        var isOpen = _props.isOpen;

        var props = _objectWithoutProperties(_props, ['isOpen']);

        var backdropClass = (0, _classnames2.default)('backdrop', {
            'in': isOpen
        });
        // e.preventDefault() in onTouchMove is used to prevent background scrolling
        return _react2.default.createElement('div', _extends({ className: backdropClass, onTouchMove: function onTouchMove(e) {
                e.preventDefault();
            } }, props));
    }
});

exports.default = Backdrop;
