// mui-react/lib/components/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _elementType = require('react-prop-types/lib/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * Button
                                                                                                                                                                                                                              * @author mayingcong <mayingcong@xingxy.cn>
                                                                                                                                                                                                                              * @date   2016-11-01
                                                                                                                                                                                                                              */

var Button = _react2.default.createClass({
    displayName: 'Button',

    propTypes: {
        active: _react2.default.PropTypes.bool,
        block: _react2.default.PropTypes.bool,
        componentClass: _elementType2.default,
        disabled: _react2.default.PropTypes.bool,
        href: _react2.default.PropTypes.string,
        onClick: _react2.default.PropTypes.func,
        type: _react2.default.PropTypes.oneOf(['button', 'submit', 'reset'])
    },
    getDefaultProps: function getDefaultProps() {
        return {
            block: false,
            active: false,
            disabled: false
        };
    },
    render: function render() {
        var _props = this.props;
        var active = _props.active;
        var disabled = _props.disabled;
        var className = _props.className;
        var href = _props.href;
        var type = _props.type;
        var block = _props.block;
        var componentClass = _props.componentClass;

        var props = _objectWithoutProperties(_props, ['active', 'disabled', 'className', 'href', 'type', 'block', 'componentClass']);

        var Component = componentClass || 'button';
        var fullClassName = (0, _classnames2.default)('btn', className, {
            'btn-block': block,
            'active': active,
            'disabled': disabled
        });
        if (href) {
            return _react2.default.createElement('a', _extends({}, props, { href: href, className: fullClassName }));
        }
        return _react2.default.createElement(Component, _extends({}, props, { type: type || 'button', className: fullClassName }));
    }
});

exports.default = Button;
