// mui-react/lib/components/ButtonGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * ButtonGroup
 * @author mayingcong <mayingcong@xingxy.cn>
 * @date   2016-11-01
 */
var ButtonGroup = _react2.default.createClass({
    displayName: 'ButtonGroup',

    propTypes: {
        size: _react2.default.PropTypes.oneOf(['xs', 'sm', 'lg', 'xl']),
        justified: _react2.default.PropTypes.bool
    },
    getDefaultProps: function getDefaultProps() {
        return {
            justified: false
        };
    },
    render: function render() {
        var _classnames;

        var _props = this.props;
        var size = _props.size;
        var justified = _props.justified;

        var props = _objectWithoutProperties(_props, ['size', 'justified']);

        var classPrefix = 'btn-group';
        var buttonGroupClass = (0, _classnames3.default)(classPrefix, (_classnames = {}, _defineProperty(_classnames, classPrefix + '-' + size, size), _defineProperty(_classnames, classPrefix + '-justified', justified), _classnames));
        // if (justified) {
        //     props.children = props.children.map((value, key) => {
        //         return (<div className={classPrefix}>{value}</div>);
        //     });
        // }
        return _react2.default.createElement('div', _extends({ className: buttonGroupClass }, props));
    }
});

exports.default = ButtonGroup;
