// mui-react/lib/components/Grid.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * @file This file exports Grid component
                                                                                                                                                                                                                              * @author mayingcong <mayingcong@xingxy.cn>
                                                                                                                                                                                                                              * @date   2016-11-01
                                                                                                                                                                                                                              */


var Grid = _react2.default.createClass({
    displayName: 'Grid',

    propTypes: {
        component: _react2.default.PropTypes.node,
        wrap: _react2.default.PropTypes.oneOf(['wrap', 'wrap-reverse']),
        align: _react2.default.PropTypes.oneOf(['right', 'center', 'between', 'around']),
        avg: _react2.default.PropTypes.number
    },
    getDefaultProps: function getDefaultProps() {
        return {
            component: 'div'
        };
    },
    render: function render() {
        var _classnames;

        var _props = this.props;
        var Component = _props.component;
        var wrap = _props.wrap;
        var align = _props.align;
        var avg = _props.avg;

        var props = _objectWithoutProperties(_props, ['component', 'wrap', 'align', 'avg']);

        var base = 'row';
        var rowClass = (0, _classnames3.default)(base, (_classnames = {}, _defineProperty(_classnames, base + '-' + wrap, wrap), _defineProperty(_classnames, base + '-' + align, align), _defineProperty(_classnames, base + '-avg-' + avg, avg), _classnames));
        return _react2.default.createElement(Component, _extends({ className: rowClass }, props));
    }
});

exports.default = Grid;
