// mui-react/lib/components/Slide.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactAddonsCssTransitionGroup = require('react-addons-css-transition-group');

var _reactAddonsCssTransitionGroup2 = _interopRequireDefault(_reactAddonsCssTransitionGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * @file This file exports a Slide component
                                                                                                                                                                                                                              * @author mayingcong <mayingcong@xingxy.cn>
                                                                                                                                                                                                                              * @deprecated  2016-11-03
                                                                                                                                                                                                                              */

var TRANSITION_TIME = 300;
var Slide = _react2.default.createClass({
    displayName: 'Slide',
    render: function render() {
        var _props = this.props;
        var transition = _props.transition;

        var props = _objectWithoutProperties(_props, ['transition']);

        return _react2.default.createElement(_reactAddonsCssTransitionGroup2.default, _extends({
            component: 'div',
            className: 'slide-container',
            transitionName: transition,
            transitionEnterTimeout: TRANSITION_TIME,
            transitionLeaveTimeout: TRANSITION_TIME
        }, props));
    }
});

var SlideItem = _react2.default.createClass({
    displayName: 'SlideItem',
    render: function render() {
        return _react2.default.createElement('div', _extends({ className: 'slide' }, this.props));
    }
});

Slide.Item = SlideItem;

exports.default = Slide;
