// mui-react/es/components/Backdrop.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
/**
 * @file This file contains Backdrop component, which is used in Modal and Picker
 * @author mayingcong <mayingcong@xingxy.cn>
 * @date 2016-11-10
 */

import React from 'react';
import classnames from 'classnames';

var Backdrop = React.createClass({
    displayName: 'Backdrop',

    propTypes: {
        isOpen: React.PropTypes.bool
    },
    getDefaultProps: function getDefaultProps() {
        return {
            isOpen: false
        };
    },

    render: function render() {
        var _props = this.props;
        var isOpen = _props.isOpen;

        var props = _objectWithoutProperties(_props, ['isOpen']);

        var backdropClass = classnames('backdrop', {
            'in': isOpen
        });
        // e.preventDefault() in onTouchMove is used to prevent background scrolling
        return React.createElement('div', _extends({ className: backdropClass, onTouchMove: function onTouchMove(e) {
                e.preventDefault();
            } }, props));
    }
});

export default Backdrop;
