// mui-react/es/components/ButtonGroup.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import React from 'react';
import classnames from 'classnames';

/**
 * ButtonGroup
 * @author mayingcong <mayingcong@xingxy.cn>
 * @date   2016-11-01
 */
var ButtonGroup = React.createClass({
    displayName: 'ButtonGroup',

    propTypes: {
        size: React.PropTypes.oneOf(['xs', 'sm', 'lg', 'xl']),
        justified: React.PropTypes.bool
    },
    getDefaultProps: function getDefaultProps() {
        return {
            justified: false
        };
    },
    render: function render() {
        var _classnames;

        var _props = this.props;
        var size = _props.size;
        var justified = _props.justified;

        var props = _objectWithoutProperties(_props, ['size', 'justified']);

        var classPrefix = 'btn-group';
        var buttonGroupClass = classnames(classPrefix, (_classnames = {}, _classnames[classPrefix + '-' + size] = size, _classnames[classPrefix + '-justified'] = justified, _classnames));
        // if (justified) {
        //     props.children = props.children.map((value, key) => {
        //         return (<div className={classPrefix}>{value}</div>);
        //     });
        // }
        return React.createElement('div', _extends({ className: buttonGroupClass }, props));
    }
});

export default ButtonGroup;
