// mui-react/es/components/Grid.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
/**
 * @file This file exports Grid component
 * @author mayingcong <mayingcong@xingxy.cn>
 * @date   2016-11-01
 */
import React from 'react';
import classnames from 'classnames';

var Grid = React.createClass({
    displayName: 'Grid',

    propTypes: {
        component: React.PropTypes.node,
        wrap: React.PropTypes.oneOf(['wrap', 'wrap-reverse']),
        align: React.PropTypes.oneOf(['right', 'center', 'between', 'around']),
        avg: React.PropTypes.number
    },
    getDefaultProps: function getDefaultProps() {
        return {
            component: 'div'
        };
    },
    render: function render() {
        var _classnames;

        var _props = this.props;
        var Component = _props.component;
        var wrap = _props.wrap;
        var align = _props.align;
        var avg = _props.avg;

        var props = _objectWithoutProperties(_props, ['component', 'wrap', 'align', 'avg']);

        var base = 'row';
        var rowClass = classnames(base, (_classnames = {}, _classnames[base + '-' + wrap] = wrap, _classnames[base + '-' + align] = align, _classnames[base + '-avg-' + avg] = avg, _classnames));
        return React.createElement(Component, _extends({ className: rowClass }, props));
    }
});

export default Grid;
