// mui-react/es/components/List.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
/**
 * List
 * @author mayingcong <mayingcong@xingxy.cn>
 * @date   2016-11-01
 */

import React from 'react';
import classnames from 'classnames';

var List = React.createClass({
    displayName: 'List',

    propTypes: {
        classPrefix: React.PropTypes.string,
        borderPadding: React.PropTypes.oneOf(['none', 'left', 'right', 'both'])
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'list-group',
            borderPadding: 'none'
        };
    },
    render: function render() {
        var _classnames;

        var _props = this.props;
        var classPrefix = _props.classPrefix;
        var borderPadding = _props.borderPadding;
        var props = _objectWithoutProperties(_props, ['classPrefix', 'borderPadding']);
        var listGroupClass = classnames(classPrefix, (_classnames = {}, _classnames['border-padding-' + borderPadding] = borderPadding !== 'none', _classnames));
        return React.createElement('div', _extends({ className: listGroupClass }, props));
    }
});

var ListItem = React.createClass({
    displayName: 'ListItem',

    propTypes: {
        classPrefix: React.PropTypes.string,
        title: React.PropTypes.node,
        subtitle: React.PropTypes.node,
        content: React.PropTypes.node,
        right: React.PropTypes.node,
        mid: React.PropTypes.node,
        href: React.PropTypes.string,
        shouldCenter: React.PropTypes.bool,
        color: React.PropTypes.string,
        media: React.PropTypes.node
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'list-group-item',
            shouldCenter: false
        };
    },

    /**
     * Render left component
     * @param  {React.ProtoTypes.node} media 
     * @param  {React.ProtoTypes.node} title 
     * @return {Object}      
     */
    renderLeft: function renderLeft(media, title) {
        var classPrefix = this.props.classPrefix;
        if (media) {
            return React.createElement(
                'div',
                { className: classPrefix + '-left' },
                media
            );
        } else if (title) {
            return React.createElement(
                'div',
                { className: classPrefix + '-left' },
                title
            );
        } else {
            return null;
        }
    },


    /**
     * Render right component
     * @param  {React.ProtoTypes.node} media    
     * @param  {React.ProtoTypes.node} title    
     * @param  {React.ProtoTypes.node} subtitle 
     * @param  {React.ProtoTypes.node} content  
     * @param  {React.ProtoTypes.node} right    
     * @return {Object}       
     */
    renderRight: function renderRight(media, title, subtitle, content, right) {
        var classPrefix = this.props.classPrefix;
        if (media) {
            return React.createElement(
                'div',
                { className: classPrefix + '-right-expand' },
                React.createElement(
                    'div',
                    { className: classPrefix + '-title-row' },
                    React.createElement(
                        'div',
                        { className: classPrefix + '-title' },
                        title
                    ),
                    right
                ),
                subtitle ? React.createElement(
                    'div',
                    { className: classPrefix + '-subtitle' },
                    subtitle
                ) : null,
                content ? React.createElement(
                    'div',
                    { className: classPrefix + '-content' },
                    content
                ) : null
            );
        } else if (right) {
            return React.createElement(
                'div',
                { className: classPrefix + '-right' },
                right
            );
        } else {
            return null;
        }
    },


    /**
     * Render middle component
     * @param {Object} mid Component you want to render into the middle of the List
     * @param {Boolean} shouldCenter Whether the mid component should be put in the right center of the List
     * @return {Object} Wrapped middle component
     */
    renderMid: function renderMid(mid, shouldCenter) {
        var classPrefix = this.props.classPrefix;
        if (!mid) {
            return null;
        } else if (shouldCenter) {
            // fuck center requirement
            return React.createElement(
                'div',
                { className: classPrefix + '-center' },
                React.createElement(
                    'div',
                    null,
                    mid
                )
            );
        } else {
            return React.createElement(
                'div',
                { className: classPrefix + '-mid' },
                mid
            );
        }
    },

    render: function render() {
        var _classnames2;

        var _props2 = this.props;
        var title = _props2.title;
        var subtitle = _props2.subtitle;
        var content = _props2.content;
        var right = _props2.right;
        var mid = _props2.mid;
        var media = _props2.media;
        var classPrefix = _props2.classPrefix;
        var href = _props2.href;
        var shouldCenter = _props2.shouldCenter;
        var color = _props2.color;
        var props = _objectWithoutProperties(_props2, ['title', 'subtitle', 'content', 'right', 'mid', 'media', 'classPrefix', 'href', 'shouldCenter', 'color']);
        var itemClassName = classnames(classPrefix, (_classnames2 = {}, _classnames2[classPrefix + '-' + color] = color, _classnames2));
        var Component = 'div';

        if (href) {
            Component = 'a';
            props.href = href;
        }
        delete props.children;
        return React.createElement(
            Component,
            _extends({ className: itemClassName }, props),
            this.renderLeft(media, title),
            this.renderMid(mid, shouldCenter),
            this.renderRight(media, title, subtitle, content, right)
        );
    }
});

List.Item = ListItem;
export default List;
