// mui-react/es/components/Modal.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
/**
 * @file This file contains Modal Component
 * @author mayingcong <mayingcong@xingxy.cn>
 * @date 2016-11-04
 */
import React from 'react';
import { unmountComponentAtNode, unstable_renderSubtreeIntoContainer as renderSubtreeIntoContainer } from 'react-dom';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import classnames from 'classnames';

import Backdrop from './Backdrop';
import Button from './Button';
import ButtonGroup from './ButtonGroup';

function noop() {}
/**
 * Modal component
 */
export var Modal = React.createClass({
    displayName: 'Modal',

    propTypes: {
        isOpen: React.PropTypes.bool,
        type: React.PropTypes.string,
        title: React.PropTypes.node,
        footer: React.PropTypes.node,
        closeByBackdrop: React.PropTypes.bool,
        classPrefix: React.PropTypes.string,
        onDismissed: React.PropTypes.func,
        onClosed: React.PropTypes.func,
        boxShadow: React.PropTypes.bool
    },
    getDefaultProps: function getDefaultProps() {
        return {
            boxShadow: true,
            closeByBackdrop: true,
            classPrefix: 'modal',
            onDismissed: noop,
            onClosed: noop
        };
    },
    getInitialState: function getInitialState() {
        return {
            isOpen: this.props.isOpen
        };
    },
    componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
        var isOpen = this.state.isOpen;
        if (!isOpen && nextProps.isOpen) {
            this.openModal();
        } else if (isOpen && !nextProps.isOpen) {
            this.closeModal();
        }
    },
    onDismissed: function onDismissed() {
        var _this = this;

        var shouldClose = this.props.onDismissed();
        if (shouldClose === undefined || shouldClose && !shouldClose.then) {
            this.closeModal();
        } else if (shouldClose && shouldClose.then) {
            shouldClose.then(function (data) {
                if (data) {
                    _this.closeModal();
                }
            });
        }
        console.log('bool ' + shouldClose);
    },
    onClosed: function onClosed() {
        var _this2 = this;

        var shouldClose = this.props.onClosed();
        if (shouldClose === undefined || shouldClose && !shouldClose.then) {
            this.closeModal();
        } else if (shouldClose && shouldClose.then) {
            shouldClose.then(function (data) {
                if (data) {
                    _this2.closeModal();
                }
            });
        }
        console.log('bool ' + shouldClose);
    },


    /**
     * Open Modal
     */
    openModal: function openModal() {
        if (!this.state.isOpen) this.setState({
            isOpen: true
        });
    },


    /**
     * Close Modal
     */
    closeModal: function closeModal() {
        if (this.state.isOpen) {
            this.setState({
                isOpen: false
            });
        }
    },
    renderBackdrop: function renderBackdrop() {
        var closeByBackdrop = this.props.closeByBackdrop;
        var isOpen = this.state.isOpen;

        if (closeByBackdrop) {
            return React.createElement(Backdrop, { isOpen: isOpen, onClick: this.onDismissed });
        } else {
            return React.createElement(Backdrop, { isOpen: isOpen });
        }
    },
    render: function render() {
        var isOpen = this.state.isOpen;
        var _props = this.props;
        var title = _props.title;
        var children = _props.children;
        var type = _props.type;
        var footer = _props.footer;
        var boxShadow = _props.boxShadow;
        var classPrefix = _props.classPrefix;

        var modalClass = classnames(classPrefix, {
            'box-shadow': boxShadow,
            'in': isOpen
        });
        return React.createElement(
            ReactCSSTransitionGroup,
            {
                component: 'div',
                transitionName: classPrefix + '-transition',
                transitionEnterTimeout: 400,
                transitionLeaveTimeout: 400
            },
            this.state.isOpen ? React.createElement(
                'div',
                { onTouchMove: function onTouchMove(e) {
                        e.preventDefault();
                    }, key: 'modal' },
                React.createElement(
                    'div',
                    { className: modalClass },
                    React.createElement(
                        Header,
                        { classPrefix: classPrefix },
                        title
                    ),
                    React.createElement(
                        Body,
                        { classPrefix: classPrefix },
                        children
                    ),
                    React.createElement(Footer, { classPrefix: classPrefix, footer: footer, type: type, onDismissed: this.onDismissed, onClosed: this.onClosed })
                ),
                this.renderBackdrop()
            ) : React.createElement('span', { key: 'none' })
        );
    }
});

/**
 * Modal.Header
 */
var Header = React.createClass({
    displayName: 'Header',

    propTypes: {
        classPrefix: React.PropTypes.string
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'modal'
        };
    },
    render: function render() {
        var _props2 = this.props;
        var classPrefix = _props2.classPrefix;

        var props = _objectWithoutProperties(_props2, ['classPrefix']);

        return React.createElement('div', _extends({ className: classPrefix + '-header' }, props));
    }
});

/**
 * Modal.Body
 */
var Body = React.createClass({
    displayName: 'Body',

    propTypes: {
        classPrefix: React.PropTypes.string
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'modal'
        };
    },
    render: function render() {
        var _props3 = this.props;
        var children = _props3.children;
        var classPrefix = _props3.classPrefix;

        return React.createElement(
            'div',
            { className: classPrefix + '-body' },
            children
        );
    }
});

/**
 * Modal.Footer
 */
var Footer = React.createClass({
    displayName: 'Footer',

    propTypes: {
        classPrefix: React.PropTypes.string,
        type: React.PropTypes.string,
        footer: React.PropTypes.node,
        onClosed: React.PropTypes.func,
        onDismissed: React.PropTypes.func
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'modal',
            type: 'alert'
        };
    },
    render: function render() {
        var _props4 = this.props;
        var type = _props4.type;
        var classPrefix = _props4.classPrefix;
        var footer = _props4.footer;
        var onDismissed = _props4.onDismissed;
        var onClosed = _props4.onClosed;

        if (!footer) {
            if (type === 'alert') {
                footer = React.createElement(
                    Button,
                    { block: true, style: { marginBottom: '0' }, onClick: onClosed },
                    '\u786E\u5B9A'
                );
            } else if (type === 'confirm') {
                footer = React.createElement(
                    ButtonGroup,
                    { justified: true },
                    React.createElement(
                        Button,
                        { onClick: onClosed },
                        '\u662F'
                    ),
                    React.createElement(
                        Button,
                        { onClick: onDismissed },
                        '\u5426'
                    )
                );
            } else if (type === 'dialog') {
                footer = React.createElement(
                    ButtonGroup,
                    { justified: true },
                    React.createElement(
                        Button,
                        { onClick: onClosed },
                        '\u786E\u5B9A'
                    ),
                    React.createElement(
                        Button,
                        { onClick: onDismissed },
                        '\u53D6\u6D88'
                    )
                );
            }
        }
        return React.createElement(
            'div',
            { className: classPrefix + '-footer' },
            footer
        );
    }
});

/**
 * ModalPortal Component
 * This is used to wrap the Modal component and not like Modal, ModalPortal will 
 * be rendered into the body-element.
 * This is useful on IOS, since position:fixed will be disturbed by overflow:hidden
 */
var ModalPortal = React.createClass({
    displayName: 'ModalPortal',

    propTypes: {
        isOpen: React.PropTypes.bool
    },
    getDefaultProps: function getDefaultProps() {
        return {
            isOpen: false
        };
    },
    componentDidMount: function componentDidMount() {
        this.node = document.createElement('div');
        this.node.className = 'modal-portal';
        document.body.appendChild(this.node);
        this.renderModal(this.props);
    },
    componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
        this.renderModal(nextProps);
    },


    /**
     * remove this node both from dom and react-dom
     */
    componentWillUnmount: function componentWillUnmount() {
        unmountComponentAtNode(this.node);
        document.body.removeChild(this.node);
    },
    renderModal: function renderModal(props) {
        this.portal = renderSubtreeIntoContainer(this, React.createElement(Modal, props), this.node);
    },
    render: function render() {
        return null;
    }
});

Modal.Header = Header;
Modal.Body = Body;
Modal.Footer = Footer;

export default ModalPortal;
