// mui-react/es/components/Slide.js
import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
/**
 * @file This file exports a Slide component
 * @author mayingcong <mayingcong@xingxy.cn>
 * @deprecated  2016-11-03
 */

import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

var TRANSITION_TIME = 300;
var Slide = React.createClass({
    displayName: 'Slide',
    render: function render() {
        var _props = this.props;
        var transition = _props.transition;

        var props = _objectWithoutProperties(_props, ['transition']);

        return React.createElement(ReactCSSTransitionGroup, _extends({
            component: 'div',
            className: 'slide-container',
            transitionName: transition,
            transitionEnterTimeout: TRANSITION_TIME,
            transitionLeaveTimeout: TRANSITION_TIME
        }, props));
    }
});

var SlideItem = React.createClass({
    displayName: 'SlideItem',
    render: function render() {
        return React.createElement('div', _extends({ className: 'slide' }, this.props));
    }
});

Slide.Item = SlideItem;

export default Slide;
